osf/utils/datetime_aware_jsonfield.py

```diff
@@ -84,10 +86,9 @@ def formfield(self, **kwargs):
         defaults.update(kwargs)
         return super(DateTimeAwareJSONField, self).formfield(**defaults)
 
-    def from_db_value(self, value, expression, connection, context):
-        if value is None:
-            return None
-        return super(DateTimeAwareJSONField, self).to_python(decode_datetime_objects(value))
+    def from_db_value(self, value, expression, connection):
+        value = super(DateTimeAwareJSONField, self).from_db_value(value, None, None)
```

Comment:
`django.db.models.JSONField` has a well written `.from_db_value()` that we can use here. It takes care of a few checks (Null, Empty, Type, etc) and handles encoding/decoding. The only customization we have is the next line: `decode_datetime_objects(value)`. **_No exceptionh handling is added to preserver the pre-upgrade old behavior._**